job_definitions/export_data.yml

```diff
@@ -0,0 +1,35 @@
+{% set environments = ['preview', 'production'] %}
+---
+{% for environment in environments %}
+- job:
+    name: "export-data-{{ environment }}"
+    display-name: "Export API model data as CSV - {{ environment }}"
+    project-type: freestyle
+    description: "Runs the get-model-data.py script and saves the generated CSV files to Google Drive"
+    triggers:
+      - timed: "H 3 * * 1"
+    publishers:
+      - trigger-parameterized-builds:
+          - project: notify-slack
+            condition: UNSTABLE_OR_WORSE
+            predefined-parameters: |
+              USERNAME=export-data
+              JOB=Export data {{ environment }}
+              ICON=:foi:
+              STAGE={{ environment }}
+              STATUS=FAILED
+              URL=<${BUILD_URL}consoleFull|#${BUILD_NUMBER}>
+              CHANNEL=#dm-release
+    builders:
+      - shell: |
+
+          # We need to recreate the directory manually since the one created by Docker
+          # when mounting the volume will otherwise be owned by the root user.
+          rm -rf ./data && mkdir data
+
+          docker run --user $(id -u) --volume $(pwd)/data:/app/data digitalmarketplace/scripts scripts/get-model-data.py '{{ environment }}' "$DM_DATA_API_TOKEN_{{ environment|upper }}"
+
+          {% if environment == "production" %}
+          /usr/local/bin/gdrive --config "/var/lib/jenkins/.gdrive" sync upload --delete-extraneous ./data "{{ jenkins_gdrive_csv_data_export_folder_id }}"{% endif %}
```

Comment:
I ran into an annoying Jinja behaviour where it indented all other jobs in the for loop, so that it only ended up creating the production one in the end. But I think it's the empty line after `{% endif %}` that fixes it, not the endif indentation itself, so I'll move it down.